packages/xo-server/src/xo-mixins/pool.js

```diff
@@ -81,4 +81,35 @@ export default class Pools {
       await _xo.mergeXenPools(source._xapiId, target._xapiId, force)
     }
   }
+
+  async listPoolsMatchingCriteria({ minCpus = 0, minMemory = 0, minSrSize = 0, poolTag, srTag }) {
+    const hostsByPool = {}
+    const srsByPool = {}
+    const pools = []
+    const objects = this._xo.getObjects()
```

Comment:
Also, I'm not sure it makes sense to use this method which construct an intermediary collection of all the objects (with optional filtering and limitation).

If we only  want to iterate, maybe we should simply expose `Xo#_objects`.